### DISCUSSION TITLE
Play file on multiple speakers

### DISCUSSION THREAD

**Discussion by TristanBrotherton** [avantrec/soco-cli#40] [General]
I'd like to play a file on multiple speakers. Do I have to create a group, and then ungroup the speakers? Or is there another syntax for:
`sonos "office" "bedroom" play_file "my/file.mp3"`

**Comment by pwt** [comment]
Hi:

Yes, you need to create a group:

`sonos office group bedroom : bedroom play_file my/file.mp3 : office ungroup`

Note that you can send the `play_file` and `ungroup` actions to either `bedroom` or `office`; SoCo-CLI will make sure the command is sent to the group coordinator (`bedroom` in this case).

**Comment by TristanBrotherton** [comment]
Thank you for the very quick response. In case this comes up in a search for someone else. I found an amazing example that will do what I exactly what I want, which would include restoring the state of the speakers after I've played my file:

https://github.com/SoCo/SoCo/blob/master/examples/snapshot/multi_zone_snap.py

**Reply by pwt** [reply]
I'll add SoCo snapshots to SoCo-CLI at some point. (Don't know why I haven't already done so, but there's probably a reason that I've forgotten!)

**Comment by mdetweiler** [comment]
Sorry to bring back an old thread, but you explain what that command actually does?  Does it create a group and then remove "office" from it, after playing the audio file?
`sonos office group bedroom : bedroom play_file my/file.mp3 : office ungroup`

**Reply by pwt** [reply]
These are three individual `sonos` commands, separated by the ` : ` command separator.
1. Group the 'office' speaker with the 'bedroom' speaker. The 'Bedroom' speaker is the group coordinator.
2. Play a local MP3 file on the 'bedroom' speaker, which will also play on the 'office' speaker since it's grouped. This command will not conclude until the end of file playback.
3. Remove the 'office' speaker from its group.

**Comment by mdetweiler** [comment]
Thank you for the detailed explanation!
